roxie/udplib/udptrr.cpp

```diff
@@ -325,7 +326,11 @@ class CReceiveManager : implements IReceiveManager, public CInterface
         Owned<ISocket> flow_socket;
         const unsigned flow_port;
         const unsigned maxSlotsPerSender;
+        const unsigned maxPermits = 0;
         std::atomic<bool> running = { false };
+        unsigned permitsIssued = 0;
+        inline void returnPermits(unsigned &permits) { permitsIssued -= permits; permits = 0; }
```

Comment:
setting to zero is a bit confusing - at least one call also clears it afterwards.  I think better for the caller to explicitly clear